### PR TITLE
add rule R4006  XmsCodeGenerationSettingDeprecated   

### PR DESCRIPTION
**Pull request by jianyexi**

### CODE REVIEW

File: src/typescript/azure-openapi-validator/rules/XmsCodeGenerationSettingDeprecated.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+export const XmsCodeGenerationSettingDeprecated: string = "XmsCodeGenerationSettingDeprecated";
+rules.push({
+  id: "R4006",
+  name: XmsCodeGenerationSettingDeprecated,
+  severity: "warning",
+  category: "ARMViolation",
```
Comment (lmazuel) [inline src/typescript/azure-openapi-validator/rules/XmsCodeGenerationSettingDeprecated.ts]: It's not just ARM but everything autorest related

File: src/typescript/azure-openapi-validator/rules/XmsCodeGenerationSettingDeprecated.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm,
```
Comment (lmazuel) [inline src/typescript/azure-openapi-validator/rules/XmsCodeGenerationSettingDeprecated.ts]: It's not just ARM but everything autorest related

Comment (jianyexi) [reply src/typescript/azure-openapi-validator/rules/XmsCodeGenerationSettingDeprecated.ts]: thanks , it's really useful ,currently, we have a  category named SDK which means the client side violation , I think this category may be more fit for this rule.
and for openapiType ,  we only have two type ( arm & data-plane)  , do you mean to apply both of them ?

Comment (lmazuel) [reply src/typescript/azure-openapi-validator/rules/XmsCodeGenerationSettingDeprecated.ts]: Yes, I mean this applies to any Swagger whatever their purpose. The exact way to declare that, I trust you I don't know your tool :). But this should apply to everything.

**Pull request merged**